packages/server-wallet/src/protocols/state.ts

```diff
@@ -55,62 +54,42 @@ type SignedByMe = {latestSignedByMe: SignedStateWithHash};
 export type ChannelStateWithMe = ChannelState & SignedByMe;
 export type ChannelStateWithSupported = ChannelState & SignedByMe & WithSupported;
 
-export type Stage = 'Missing' | 'PrefundSetup' | 'PostfundSetup' | 'Running' | 'Final';
 /**
- *
- * @param state: SignedStateWithHash | undefined
- * Returns stage of state.
- *
- * Useful for partitioning the protocol state to decide what action to next take.
+ * The definition of a ChannelStatus is still under debate, but for now
+ * this is what we have settled on. Future changes should include 'funding',
+ * 'defunding', and state related to challenging, probably. See this issue
+ * for more: https://github.com/statechannels/statechannels/issues/2509
  */
-export const stage = (state: State | undefined): Stage =>
-  !state
-    ? 'Missing'
-    : state.isFinal
-    ? 'Final'
-    : state.turnNum <= state.participants.length - 1
-    ? 'PrefundSetup'
-    : state.turnNum <= state.participants.length * 2 - 1
-    ? 'PostfundSetup'
-    : 'Running';
+export const status = (channelState: ChannelState): ChannelStatus => {
+  const {supported, latest, latestSignedByMe, support} = channelState;
+  const {participants} = supported ?? latest;
 
-export const toChannelResult = (channelState: ChannelState): ChannelResult => {
-  const {
-    channelId,
-    supported,
-    latest,
-    latestSignedByMe,
-    support,
-    directFundingStatus,
-    challengeStatus,
-  } = channelState;
+  if (supported?.isFinal) {
+    if (support?.every(s => s.isFinal)) return 'closed'; // the entire support chain isFinal
+    return 'closing'; // the latest state, and possibly other states, are isFinal
+  }
 
-  const {outcome, appData, turnNum, participants, appDefinition} = supported ?? latest;
+  if (latest.turnNum < 2 * participants.length) {
+    if (supported?.turnNum === participants.length - 1) return 'running'; // 3
+    if (latestSignedByMe) return 'opening'; // 0 or 1 signed by me
+    return 'proposed'; // 0 or 1 signed, but not by me
+  }
 
-  const status: ChannelStatus = ((): ChannelStatus => {
-    switch (stage(supported)) {
-      case 'Missing':
-      case 'PrefundSetup':
-        return latestSignedByMe ? 'opening' : 'proposed';
-      case 'PostfundSetup':
-        return (supported?.turnNum || 0) < participants.length * 2 - 1 ? 'opening' : 'running';
-      case 'Running':
-        return 'running';
-      case 'Final':
-        return support?.find(s => !s.isFinal) ? 'closing' : 'closed';
-    }
-  })();
-
-  const allocations = serializeAllocation(checkThat(outcome, isAllocation));
+  return 'running'; // unambiguously running e.g., 4, 5, 6, ...
```

Comment:
Alice might have signed state `0` only. If Bob sends Alice state 4, this line would indicate to her app that the channel is now "running", right?

Reply:
Yeah — let me write another variant that is simpler and handles this